folding-schemes/src/folding/nova/nifs/mova_matrix.rs

```diff
@@ -0,0 +1,580 @@
+            rE_prime,
+        };
+        Ok((
+            w,
+            ci,
+            proof,
+            vec![], // r_bits, returned to be passed as inputs to the circuit, not used at the
+                    // current impl status
+        ))
+    }
+
+    /// It verifies the results from the proof
+    /// Both the folding and the pt-vs-line proof
+    /// returns the folded committed instance
+    fn verify(
+        transcript: &mut T,
+        pp_hash: C::ScalarField,
+        simple_instance: &RelaxedCommittedRelation<C>,
+        acc_instance: &RelaxedCommittedRelation<C>,
+        proof: &Proof<C>,
+    ) -> Result<(Self::CommittedInstance, Vec<bool>), Error> {
+        transcript.absorb(&pp_hash);
+        transcript.absorb(simple_instance);
+        transcript.absorb(acc_instance);
+
+        // Verify rE_prime
+        let rE_prime = PointVsLineMatrix::<C, T>::verify(
+            transcript,
+            None,
+            acc_instance,
+            &proof.h_proof,
+            None,
+            &proof.mleE2_prime,
+            &proof.rE_prime,
+        )?;
+
+        // Derive alpha
+        transcript.absorb(&proof.mleE2_prime);
+        transcript.absorb(&proof.mleT);
+        let alpha: C::ScalarField = transcript.get_challenge();
+
+        Ok((
+            Self::fold_committed_instance(
+                alpha,
+                simple_instance,
+                acc_instance,
+                &rE_prime,
+                &proof.mleE2_prime,
+                &proof.mleT,
+            )?,
+            vec![],
+        ))
+    }
+}
+
+impl<C: Curve, CS: CommitmentScheme<C, H>, T: Transcript<C::ScalarField>, const H: bool>
+    NIFS<C, CS, T, H>
+{
+    /// Folds two committed instances into a single one using the provided parameters
+    /// Protocol 5 - Step 7 (page 25) for folding instances
+    /// # Parameters
+    /// * `alpha` - Random challenge used for folding
+    /// * `simple_instance` - The simple (unfolded) instance
+    /// * `acc_instance` - The accumulated (previously folded) instance  
+    /// * `rE_prime` - New random evaluation point
+    /// * `mleE2_prime` - Evaluation of MLE[E2] at rE_prime
+    /// * `mleT` - Evaluation of the crossterm T
+    fn fold_committed_instance(
+        alpha: C::ScalarField,
+        simple_instance: &RelaxedCommittedRelation<C>,
+        acc_instance: &RelaxedCommittedRelation<C>,
+        rE_prime: &[C::ScalarField],
+        mleE2_prime: &C::ScalarField, // v' in Protocol 5
+        mleT: &C::ScalarField,
+    ) -> Result<RelaxedCommittedRelation<C>, Error> {
+        // Step 7
+        // Accumulate commitments
+        let com_a_acc = simple_instance.cmA.mul(alpha) + acc_instance.cmA;
+        let com_b_acc = simple_instance.cmB.mul(alpha) + acc_instance.cmB;
+        let com_c_acc = simple_instance.cmC.mul(alpha) + acc_instance.cmC;
+
+        // Update scalars
+        let u_acc = alpha + acc_instance.u;
+        let mlE = *mleE2_prime + alpha * *mleT;
```

Comment:
Fix below where you return mleE too
```suggestion
        let mleE = *mleE2_prime + alpha * *mleT;
```